candle-book/src/guide/hello_world.md

```diff
@@ -1 +1,53 @@
-# PyTorch cheatsheet
+# Hello world !
+
+We will now create the hello world of the ML world, building a model capable of solving MNIST dataset.
+
+Open `src/main.rs` and fill in with these contents:
+
+```rust
+# extern crate candle;
+use candle::{DType, Device, Result, Tensor};
+
+struct Model {
+    first: Tensor,
```

Comment:
Maybe it makes more sense to use `candle_nn::Linear` here?
Or maybe we really want to go back to basics here in which case we should mention that `Linear` exists but for the sake of this tutorial we won't use it here and have explicit tensors instead.

Reply:
Great idea. I was wondering how to complexify this example without delving into too much details to keep it hello world like.

I will keep this example that simple, and expand by making your own Linear, then referring to candle_nn for layers in general.

For `Conv1d` and `Linear` for instance, wdyt ?

Reply:
BTW, this comment is more about https://github.com/LaurentMazare/candle/pull/258 (This PR I converted to draft because it's more about the `save/load` surface to get simple cheatsheet more complete on that front.

Reply:
Sounds good, you could even even not have backprop/gradient descent for a tensor 101 though maybe it's too simplistic. But certainly good to only introduce `candle_nn` later in the process (so that users can understand that there is no magic behind the hood there).

Reply:
Yup, I want to keep the training loop in a different place.

IMO after having hello world + there's no magic you have several options:

- Run a real model (90% of people)
- Train a model (10% of people) 

It's an old stat, we're closer to 99% vs 1% now that AI is much more mainstream (but there's still a lot of finetuners out there, mostly using script already written by people)